PDA 1.6.0 libpdal_base.so undefined symbol when used in  CloudCompare

## Current Behavior for CloudCompare compile :1st_place_medal: 
 > make failed with the following message : 
 > [ 74%] Linking CXX executable CloudCompare
> /usr/local/lib/libpdal_base.so : référence indéfinie vers « curl_easy_reset »
> /usr/local/lib/libpdal_base.so : référence indéfinie vers « curl_slist_free_all »
 > /usr/local/lib/libpdal_base.so : référence indéfinie vers « curl_easy_perform »
> /usr/local/lib/libpdal_base.so : référence indéfinie vers « curl_slist_append »
> /usr/local/lib/libpdal_base.so : référence indéfinie vers « curl_easy_cleanup »
> /usr/local/lib/libpdal_base.so : référence indéfinie vers « curl_global_init »
 > /usr/local/lib/libpdal_base.so : référence indéfinie vers « curl_easy_setopt »
> /usr/local/lib/libpdal_base.so : référence indéfinie vers « curl_easy_getinfo »
 > /usr/local/lib/libpdal_base.so : référence indéfinie vers « curl_easy_init »
 > collect2: error: ld returned 1 exit status 
> qCC/CMakeFiles/CloudCompare.dir/build.make:2785 : la recette pour la cible « qCC/CloudCompare » a échouée
> make[2]: *** [qCC/CloudCompare] Erreur 1
> CMakeFiles/Makefile2:1147 : la recette pour la cible « qCC/CMakeFiles/CloudCompare.dir/all » a échouée
> make[1]: *** [qCC/CMakeFiles/CloudCompare.dir/all] Erreur 2
> Makefile:127 : la recette pour la cible « all » a échouée
> make: *** [all] Erreur 2

<!--- If suggesting a change/improvement, explain the difference from current behavior -->
PDAL works correctly with *.las capacity
I added path to shared an static library with LGD_LIBRARY_PATH in /etc/environment and in /etc/ld.so.conf with ldconfig activation, I create symbolic  link from /usr/local/lib to /usr/lib/x86-64-linux-gnu to libcurl.so & libcurl.a

## Expected Behavior
libcurl3 and libcurl3-dev correctly installed
Without PDAL_LAS of course no problem compile goes ahead and CloudCompare starts
with PDAL_LAS compile fails

## Steps to Reproduce (for bugs)
Configure and Generate with CMAKE 3.5.1



## Your Environment
-  1.6.0 (git-version: e4573b)
* CC Version last one from GITHUB 2.10 alpha, compile with Qt 5.7: 
* UBUNTU 16.04 LTS linux 4.10 gcc g++ 5.4.0 
* NVIDIA K5000

I am lost help needed Thank you for any hint

It would help to know how your version of PDAL is linked.  It would help if
you could provide the output of:

ldd libpdal_base.so | grep curl

On Wed, Feb 28, 2018 at 5:50 AM, Marcel Bariou <notifications@github.com>
wrote:

> Current Behavior for CloudCompare compile 🥇
>
> make failed with the following message :
> [ 74%] Linking CXX executable CloudCompare
> /usr/local/lib/libpdal_base.so : référence indéfinie vers «
> curl_easy_reset »
> /usr/local/lib/libpdal_base.so : référence indéfinie vers «
> curl_slist_free_all »
> /usr/local/lib/libpdal_base.so : référence indéfinie vers «
> curl_easy_perform »
> /usr/local/lib/libpdal_base.so : référence indéfinie vers «
> curl_slist_append »
> /usr/local/lib/libpdal_base.so : référence indéfinie vers «
> curl_easy_cleanup »
> /usr/local/lib/libpdal_base.so : référence indéfinie vers «
> curl_global_init »
> /usr/local/lib/libpdal_base.so : référence indéfinie vers «
> curl_easy_setopt »
> /usr/local/lib/libpdal_base.so : référence indéfinie vers «
> curl_easy_getinfo »
> /usr/local/lib/libpdal_base.so : référence indéfinie vers « curl_easy_init
> »
> collect2: error: ld returned 1 exit status
> qCC/CMakeFiles/CloudCompare.dir/build.make:2785 : la recette pour la
> cible « qCC/CloudCompare » a échouée
> make[2]: *** [qCC/CloudCompare] Erreur 1
> CMakeFiles/Makefile2:1147 : la recette pour la cible «
> qCC/CMakeFiles/CloudCompare.dir/all » a échouée
> make[1]: *** [qCC/CMakeFiles/CloudCompare.dir/all] Erreur 2
> Makefile:127 : la recette pour la cible « all » a échouée
> make: *** [all] Erreur 2
>
> PDAL works correctly with *.las capacity
> I added path to shared an static library with LGD_LIBRARY_PATH in
> /etc/environment and in /etc/ld.so.conf with ldconfig activation, I create
> symbolic link from /usr/local/lib to /usr/lib/x86-64-linux-gnu to
> libcurl.so & libcurl.a
> Expected Behavior
>
> libcurl3 and libcurl3-dev correctly installed
> Without PDAL_LAS of course no problem compile goes ahead and CloudCompare
> starts
> with PDAL_LAS compile fails
> Steps to Reproduce (for bugs)
>
> Configure and Generate with CMAKE 3.5.1
> Your Environment
>
>    - 1.6.0 (git-version: e4573b)
>
>
>    - CC Version last one from GITHUB 2.10 alpha, compile with Qt 5.7:
>    - UBUNTU 16.04 LTS linux 4.10 gcc g++ 5.4.0
>    - NVIDIA K5000
>
> I am lost help needed Thank you for any hint
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/PDAL/PDAL/issues/1822>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABQWMJma0fejFV2pnX10p1QwEMjGGIE6ks5tZS9XgaJpZM4SWcxg>
> .
>



-- 
Andrew Bell
andrew.bell.ia@gmail.com

I have encountered the same issue on my Gentoo system:

`[ 52%] Linking CXX executable ../../bin/xml_schema_test
cd /var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/test/unit && /usr/bin/cmake -E cmake_link_script CMakeFiles/xml_schema_test.dir/link.txt --verbose=1
/usr/x86_64-pc-linux-gnu/gcc-bin/7.3.0/x86_64-pc-linux-gnu-g++  -march=native -O2 -pipe -ggdb  -Wl,-O1 -Wl,--as-needed CMakeFiles/xml_schema_test.dir/XMLSchemaTest.cpp.o CMakeFiles/pdal_test_support.dir/Support.cpp.o  -o ../../bin/xml_schema_test -Wl,-rpath,/var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/lib64 ../../lib64/libpdal_base.so.8.0.0 ../../lib64/libpdal_util.so.8.0.0 ../../lib64/libgtest.a /usr/lib64/libxml2.so -lpthread 
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_easy_reset'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_slist_append'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_easy_perform'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_easy_init'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_global_init'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_easy_cleanup'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_easy_setopt'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_easy_getinfo'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_slist_free_all'
collect2: error: ld returned 1 exit status
make[2]: *** [test/unit/CMakeFiles/xml_schema_test.dir/build.make:101: bin/xml_schema_test] Error 1
make[2]: Leaving directory '/var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build'
make[1]: *** [CMakeFiles/Makefile2:1389: test/unit/CMakeFiles/xml_schema_test.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
`

Just as with @marcel-bariou, the libpdal_base.so is not linked against libcurl despite libcurl being being included in the linker line:

`/usr/x86_64-pc-linux-gnu/gcc-bin/7.3.0/x86_64-pc-linux-gnu-g++ -fPIC -march=native -O2 -pipe -ggdb -Wl,-O1 -Wl,--as-needed -shared -Wl,-soname,libpdal_base.so.7 -o lib64/libpdal_base.so.8.0.0 CMakeFiles/pdal_base.dir/filters/ApproximateCoplanarFilter.cpp.o CMakeFiles/pdal_base.dir/filters/AssignFilter.cpp.o CMakeFiles/pdal_base.dir/filters/ChipperFilter.cpp.o CMakeFiles/pdal_base.dir/filters/ClusterFilter.cpp.o CMakeFiles/pdal_base.dir/filters/ColorinterpFilter.cpp.o CMakeFiles/pdal_base.dir/filters/ColorizationFilter.cpp.o CMakeFiles/pdal_base.dir/filters/CropFilter.cpp.o CMakeFiles/pdal_base.dir/filters/DEMFilter.cpp.o CMakeFiles/pdal_base.dir/filters/DecimationFilter.cpp.o CMakeFiles/pdal_base.dir/filters/DividerFilter.cpp.o CMakeFiles/pdal_base.dir/filters/ELMFilter.cpp.o CMakeFiles/pdal_base.dir/filters/EigenvaluesFilter.cpp.o CMakeFiles/pdal_base.dir/filters/EstimateRankFilter.cpp.o CMakeFiles/pdal_base.dir/filters/FerryFilter.cpp.o CMakeFiles/pdal_base.dir/filters/GreedyProjection.cpp.o CMakeFiles/pdal_base.dir/filters/GroupByFilter.cpp.o CMakeFiles/pdal_base.dir/filters/HAGFilter.cpp.o CMakeFiles/pdal_base.dir/filters/HeadFilter.cpp.o CMakeFiles/pdal_base.dir/filters/HexBinFilter.cpp.o CMakeFiles/pdal_base.dir/filters/IQRFilter.cpp.o CMakeFiles/pdal_base.dir/filters/InfoFilter.cpp.o CMakeFiles/pdal_base.dir/filters/KDistanceFilter.cpp.o CMakeFiles/pdal_base.dir/filters/LOFFilter.cpp.o CMakeFiles/pdal_base.dir/filters/LocateFilter.cpp.o CMakeFiles/pdal_base.dir/filters/MADFilter.cpp.o CMakeFiles/pdal_base.dir/filters/MergeFilter.cpp.o CMakeFiles/pdal_base.dir/filters/MongusFilter.cpp.o CMakeFiles/pdal_base.dir/filters/MortonOrderFilter.cpp.o CMakeFiles/pdal_base.dir/filters/NNDistanceFilter.cpp.o CMakeFiles/pdal_base.dir/filters/NeighborClassifierFilter.cpp.o CMakeFiles/pdal_base.dir/filters/NormalFilter.cpp.o CMakeFiles/pdal_base.dir/filters/OutlierFilter.cpp.o CMakeFiles/pdal_base.dir/filters/OverlayFilter.cpp.o CMakeFiles/pdal_base.dir/filters/PMFFilter.cpp.o CMakeFiles/pdal_base.dir/filters/PoissonFilter.cpp.o CMakeFiles/pdal_base.dir/filters/RadialDensityFilter.cpp.o CMakeFiles/pdal_base.dir/filters/RandomizeFilter.cpp.o CMakeFiles/pdal_base.dir/filters/RangeFilter.cpp.o CMakeFiles/pdal_base.dir/filters/ReprojectionFilter.cpp.o CMakeFiles/pdal_base.dir/filters/ReturnsFilter.cpp.o CMakeFiles/pdal_base.dir/filters/SMRFilter.cpp.o CMakeFiles/pdal_base.dir/filters/SampleFilter.cpp.o CMakeFiles/pdal_base.dir/filters/SortFilter.cpp.o CMakeFiles/pdal_base.dir/filters/SplitterFilter.cpp.o CMakeFiles/pdal_base.dir/filters/StatsFilter.cpp.o CMakeFiles/pdal_base.dir/filters/StreamCallbackFilter.cpp.o CMakeFiles/pdal_base.dir/filters/TailFilter.cpp.o CMakeFiles/pdal_base.dir/filters/TransformationFilter.cpp.o CMakeFiles/pdal_base.dir/filters/VoxelCenterNearestNeighborFilter.cpp.o CMakeFiles/pdal_base.dir/filters/VoxelCentroidNearestNeighborFilter.cpp.o CMakeFiles/pdal_base.dir/io/BpfCompressor.cpp.o CMakeFiles/pdal_base.dir/io/BpfHeader.cpp.o CMakeFiles/pdal_base.dir/io/BpfReader.cpp.o CMakeFiles/pdal_base.dir/io/BpfWriter.cpp.o CMakeFiles/pdal_base.dir/io/EptReader.cpp.o CMakeFiles/pdal_base.dir/io/FauxReader.cpp.o CMakeFiles/pdal_base.dir/io/GDALGrid.cpp.o CMakeFiles/pdal_base.dir/io/GDALReader.cpp.o CMakeFiles/pdal_base.dir/io/GDALWriter.cpp.o CMakeFiles/pdal_base.dir/io/GeotiffSupport.cpp.o CMakeFiles/pdal_base.dir/io/Ilvis2MetadataReader.cpp.o CMakeFiles/pdal_base.dir/io/Ilvis2Reader.cpp.o CMakeFiles/pdal_base.dir/io/LasHeader.cpp.o CMakeFiles/pdal_base.dir/io/LasReader.cpp.o CMakeFiles/pdal_base.dir/io/LasSummaryData.cpp.o CMakeFiles/pdal_base.dir/io/LasUtils.cpp.o CMakeFiles/pdal_base.dir/io/LasVLR.cpp.o CMakeFiles/pdal_base.dir/io/LasWriter.cpp.o CMakeFiles/pdal_base.dir/io/NullWriter.cpp.o CMakeFiles/pdal_base.dir/io/OGRWriter.cpp.o CMakeFiles/pdal_base.dir/io/OGRWriterV1.cpp.o CMakeFiles/pdal_base.dir/io/OptechReader.cpp.o CMakeFiles/pdal_base.dir/io/PlyReader.cpp.o CMakeFiles/pdal_base.dir/io/PlyWriter.cpp.o CMakeFiles/pdal_base.dir/io/PtsReader.cpp.o CMakeFiles/pdal_base.dir/io/QfitReader.cpp.o CMakeFiles/pdal_base.dir/io/SbetReader.cpp.o CMakeFiles/pdal_base.dir/io/SbetWriter.cpp.o CMakeFiles/pdal_base.dir/io/TIndexReader.cpp.o CMakeFiles/pdal_base.dir/io/TerrasolidReader.cpp.o CMakeFiles/pdal_base.dir/io/TextReader.cpp.o CMakeFiles/pdal_base.dir/io/TextWriter.cpp.o CMakeFiles/pdal_base.dir/kernels/DeltaKernel.cpp.o CMakeFiles/pdal_base.dir/kernels/DensityKernel.cpp.o CMakeFiles/pdal_base.dir/kernels/DiffKernel.cpp.o CMakeFiles/pdal_base.dir/kernels/GroundKernel.cpp.o CMakeFiles/pdal_base.dir/kernels/HausdorffKernel.cpp.o CMakeFiles/pdal_base.dir/kernels/InfoKernel.cpp.o CMakeFiles/pdal_base.dir/kernels/MergeKernel.cpp.o CMakeFiles/pdal_base.dir/kernels/PipelineKernel.cpp.o CMakeFiles/pdal_base.dir/kernels/RandomKernel.cpp.o CMakeFiles/pdal_base.dir/kernels/SortKernel.cpp.o CMakeFiles/pdal_base.dir/kernels/SplitKernel.cpp.o CMakeFiles/pdal_base.dir/kernels/TIndexKernel.cpp.o CMakeFiles/pdal_base.dir/kernels/TileKernel.cpp.o CMakeFiles/pdal_base.dir/kernels/TranslateKernel.cpp.o CMakeFiles/pdal_base.dir/pdal/DbReader.cpp.o CMakeFiles/pdal_base.dir/pdal/DbWriter.cpp.o CMakeFiles/pdal_base.dir/pdal/DynamicLibrary.cpp.o CMakeFiles/pdal_base.dir/pdal/EigenUtils.cpp.o CMakeFiles/pdal_base.dir/pdal/GDALUtils.cpp.o CMakeFiles/pdal_base.dir/pdal/GEOSUtils.cpp.o CMakeFiles/pdal_base.dir/pdal/Geometry.cpp.o CMakeFiles/pdal_base.dir/pdal/Kernel.cpp.o CMakeFiles/pdal_base.dir/pdal/Log.cpp.o CMakeFiles/pdal_base.dir/pdal/Metadata.cpp.o CMakeFiles/pdal_base.dir/pdal/Options.cpp.o CMakeFiles/pdal_base.dir/pdal/PDALUtils.cpp.o CMakeFiles/pdal_base.dir/pdal/PipelineExecutor.cpp.o CMakeFiles/pdal_base.dir/pdal/PipelineManager.cpp.o CMakeFiles/pdal_base.dir/pdal/PipelineReaderJSON.cpp.o CMakeFiles/pdal_base.dir/pdal/PipelineWriter.cpp.o CMakeFiles/pdal_base.dir/pdal/PluginDirectory.cpp.o CMakeFiles/pdal_base.dir/pdal/PluginManager.cpp.o CMakeFiles/pdal_base.dir/pdal/PointLayout.cpp.o CMakeFiles/pdal_base.dir/pdal/PointTable.cpp.o CMakeFiles/pdal_base.dir/pdal/PointView.cpp.o CMakeFiles/pdal_base.dir/pdal/Polygon.cpp.o CMakeFiles/pdal_base.dir/pdal/QuadIndex.cpp.o CMakeFiles/pdal_base.dir/pdal/Reader.cpp.o CMakeFiles/pdal_base.dir/pdal/Scaling.cpp.o CMakeFiles/pdal_base.dir/pdal/SpatialReference.cpp.o CMakeFiles/pdal_base.dir/pdal/Stage.cpp.o CMakeFiles/pdal_base.dir/pdal/StageExtensions.cpp.o CMakeFiles/pdal_base.dir/pdal/StageFactory.cpp.o CMakeFiles/pdal_base.dir/pdal/Streamable.cpp.o CMakeFiles/pdal_base.dir/pdal/Writer.cpp.o CMakeFiles/pdal_base.dir/pdal/XMLSchema.cpp.o CMakeFiles/pdal_base.dir/pdal/gitsha.cpp.o CMakeFiles/pdal_base.dir/pdal/pdal_config.cpp.o CMakeFiles/pdal_base.dir/pdal/compression/DeflateCompression.cpp.o CMakeFiles/pdal_base.dir/filters/private/DimRange.cpp.o CMakeFiles/pdal_base.dir/filters/private/Point.cpp.o CMakeFiles/pdal_base.dir/filters/private/Segmentation.cpp.o CMakeFiles/pdal_base.dir/filters/private/hexer/HexGrid.cpp.o CMakeFiles/pdal_base.dir/filters/private/hexer/Hexagon.cpp.o CMakeFiles/pdal_base.dir/filters/private/hexer/Path.cpp.o CMakeFiles/pdal_base.dir/filters/private/hexer/Processor.cpp.o CMakeFiles/pdal_base.dir/filters/private/hexer/Segment.cpp.o CMakeFiles/pdal_base.dir/kernels/private/density/OGR.cpp.o -Wl,-rpath,/var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/lib64: -lpthread -lgdal -lgeos_c -lgeotiff -lxml2 -lz -lcurl lib64/libpdal_util.so.8.0.0 lib64/libpdal_arbiter.a lib64/libpdal_kazhdan.a -ljsoncpp 
`

His suggestion to install curlpp hasn't resolved the issue for me, not even when I push the -lcurlpp into LDFLAGS:

`[ 53%] Linking CXX executable ../../bin/pdal_io_ilvis2_reader_metadata_test
cd /var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/test/unit && /usr/bin/cmake -E cmake_link_script CMakeFiles/pdal_io_ilvis2_reader_metadata_test.dir/link.txt --verbose=1
/usr/x86_64-pc-linux-gnu/gcc-bin/7.3.0/x86_64-pc-linux-gnu-g++  -march=native -O2 -pipe -ggdb  -Wl,-O1 -Wl,--as-needed -lcurlpp CMakeFiles/pdal_io_ilvis2_reader_metadata_test.dir/io/Ilvis2ReaderWithMDReaderTest.cpp.o CMakeFiles/pdal_test_support.dir/Support.cpp.o  -o ../../bin/pdal_io_ilvis2_reader_metadata_test -Wl,-rpath,/var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/lib64 ../../lib64/libpdal_base.so.8.0.0 ../../lib64/libpdal_util.so.8.0.0 ../../lib64/libgtest.a -lpthread 
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_easy_reset'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_slist_append'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_easy_perform'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_easy_init'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_global_init'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_easy_cleanup'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_easy_setopt'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_easy_getinfo'
../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_slist_free_all'
collect2: error: ld returned 1 exit status
`

I wonder if it can be a compiler issue.

Just a hint: is your compiler looking in the right PATH ?  a dirty solution
:create  a symbolic link, in the presumed PATH to the actual LIBRARY,  to
solve in the hurry


Le lun. 3 déc. 2018 à 15:14, Dirk <notifications@github.com> a écrit :

> I have encountered the same issue on my Gentoo system:
>
> [ 52%] Linking CXX executable ../../bin/xml_schema_test cd
> /var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/test/unit
> && /usr/bin/cmake -E cmake_link_script
> CMakeFiles/xml_schema_test.dir/link.txt --verbose=1
> /usr/x86_64-pc-linux-gnu/gcc-bin/7.3.0/x86_64-pc-linux-gnu-g++
> -march=native -O2 -pipe -ggdb -Wl,-O1 -Wl,--as-needed
> CMakeFiles/xml_schema_test.dir/XMLSchemaTest.cpp.o
> CMakeFiles/pdal_test_support.dir/Support.cpp.o -o ../../bin/xml_schema_test
> -Wl,-rpath,/var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/lib64
> ../../lib64/libpdal_base.so.8.0.0 ../../lib64/libpdal_util.so.8.0.0
> ../../lib64/libgtest.a /usr/lib64/libxml2.so -lpthread
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to 'curl_easy_reset'
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to
> 'curl_slist_append' ../../lib64/libpdal_base.so.8.0.0: undefined reference
> to 'curl_easy_perform' ../../lib64/libpdal_base.so.8.0.0: undefined
> reference to 'curl_easy_init' ../../lib64/libpdal_base.so.8.0.0: undefined
> reference to 'curl_global_init' ../../lib64/libpdal_base.so.8.0.0:
> undefined reference to 'curl_easy_cleanup'
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to
> 'curl_easy_setopt' ../../lib64/libpdal_base.so.8.0.0: undefined reference
> to 'curl_easy_getinfo' ../../lib64/libpdal_base.so.8.0.0: undefined
> reference to 'curl_slist_free_all' collect2: error: ld returned 1 exit
> status make[2]: ***
> [test/unit/CMakeFiles/xml_schema_test.dir/build.make:101:
> bin/xml_schema_test] Error 1 make[2]: Leaving directory
> '/var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build'
> make[1]: *** [CMakeFiles/Makefile2:1389:
> test/unit/CMakeFiles/xml_schema_test.dir/all] Error 2 make[1]: *** Waiting
> for unfinished jobs....
>
> Just as with @marcel-bariou <https://github.com/marcel-bariou>, the
> libpdal_base.so is not linked against libcurl despite libcurl being being
> included in the linker line:
>
> /usr/x86_64-pc-linux-gnu/gcc-bin/7.3.0/x86_64-pc-linux-gnu-g++ -fPIC
> -march=native -O2 -pipe -ggdb -Wl,-O1 -Wl,--as-needed -shared
> -Wl,-soname,libpdal_base.so.7 -o lib64/libpdal_base.so.8.0.0
> CMakeFiles/pdal_base.dir/filters/ApproximateCoplanarFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/AssignFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/ChipperFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/ClusterFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/ColorinterpFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/ColorizationFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/CropFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/DEMFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/DecimationFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/DividerFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/ELMFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/EigenvaluesFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/EstimateRankFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/FerryFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/GreedyProjection.cpp.o
> CMakeFiles/pdal_base.dir/filters/GroupByFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/HAGFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/HeadFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/HexBinFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/IQRFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/InfoFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/KDistanceFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/LOFFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/LocateFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/MADFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/MergeFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/MongusFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/MortonOrderFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/NNDistanceFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/NeighborClassifierFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/NormalFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/OutlierFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/OverlayFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/PMFFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/PoissonFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/RadialDensityFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/RandomizeFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/RangeFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/ReprojectionFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/ReturnsFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/SMRFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/SampleFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/SortFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/SplitterFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/StatsFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/StreamCallbackFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/TailFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/TransformationFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/VoxelCenterNearestNeighborFilter.cpp.o
> CMakeFiles/pdal_base.dir/filters/VoxelCentroidNearestNeighborFilter.cpp.o
> CMakeFiles/pdal_base.dir/io/BpfCompressor.cpp.o
> CMakeFiles/pdal_base.dir/io/BpfHeader.cpp.o
> CMakeFiles/pdal_base.dir/io/BpfReader.cpp.o
> CMakeFiles/pdal_base.dir/io/BpfWriter.cpp.o
> CMakeFiles/pdal_base.dir/io/EptReader.cpp.o
> CMakeFiles/pdal_base.dir/io/FauxReader.cpp.o
> CMakeFiles/pdal_base.dir/io/GDALGrid.cpp.o
> CMakeFiles/pdal_base.dir/io/GDALReader.cpp.o
> CMakeFiles/pdal_base.dir/io/GDALWriter.cpp.o
> CMakeFiles/pdal_base.dir/io/GeotiffSupport.cpp.o
> CMakeFiles/pdal_base.dir/io/Ilvis2MetadataReader.cpp.o
> CMakeFiles/pdal_base.dir/io/Ilvis2Reader.cpp.o
> CMakeFiles/pdal_base.dir/io/LasHeader.cpp.o
> CMakeFiles/pdal_base.dir/io/LasReader.cpp.o
> CMakeFiles/pdal_base.dir/io/LasSummaryData.cpp.o
> CMakeFiles/pdal_base.dir/io/LasUtils.cpp.o
> CMakeFiles/pdal_base.dir/io/LasVLR.cpp.o
> CMakeFiles/pdal_base.dir/io/LasWriter.cpp.o
> CMakeFiles/pdal_base.dir/io/NullWriter.cpp.o
> CMakeFiles/pdal_base.dir/io/OGRWriter.cpp.o
> CMakeFiles/pdal_base.dir/io/OGRWriterV1.cpp.o
> CMakeFiles/pdal_base.dir/io/OptechReader.cpp.o
> CMakeFiles/pdal_base.dir/io/PlyReader.cpp.o
> CMakeFiles/pdal_base.dir/io/PlyWriter.cpp.o
> CMakeFiles/pdal_base.dir/io/PtsReader.cpp.o
> CMakeFiles/pdal_base.dir/io/QfitReader.cpp.o
> CMakeFiles/pdal_base.dir/io/SbetReader.cpp.o
> CMakeFiles/pdal_base.dir/io/SbetWriter.cpp.o
> CMakeFiles/pdal_base.dir/io/TIndexReader.cpp.o
> CMakeFiles/pdal_base.dir/io/TerrasolidReader.cpp.o
> CMakeFiles/pdal_base.dir/io/TextReader.cpp.o
> CMakeFiles/pdal_base.dir/io/TextWriter.cpp.o
> CMakeFiles/pdal_base.dir/kernels/DeltaKernel.cpp.o
> CMakeFiles/pdal_base.dir/kernels/DensityKernel.cpp.o
> CMakeFiles/pdal_base.dir/kernels/DiffKernel.cpp.o
> CMakeFiles/pdal_base.dir/kernels/GroundKernel.cpp.o
> CMakeFiles/pdal_base.dir/kernels/HausdorffKernel.cpp.o
> CMakeFiles/pdal_base.dir/kernels/InfoKernel.cpp.o
> CMakeFiles/pdal_base.dir/kernels/MergeKernel.cpp.o
> CMakeFiles/pdal_base.dir/kernels/PipelineKernel.cpp.o
> CMakeFiles/pdal_base.dir/kernels/RandomKernel.cpp.o
> CMakeFiles/pdal_base.dir/kernels/SortKernel.cpp.o
> CMakeFiles/pdal_base.dir/kernels/SplitKernel.cpp.o
> CMakeFiles/pdal_base.dir/kernels/TIndexKernel.cpp.o
> CMakeFiles/pdal_base.dir/kernels/TileKernel.cpp.o
> CMakeFiles/pdal_base.dir/kernels/TranslateKernel.cpp.o
> CMakeFiles/pdal_base.dir/pdal/DbReader.cpp.o
> CMakeFiles/pdal_base.dir/pdal/DbWriter.cpp.o
> CMakeFiles/pdal_base.dir/pdal/DynamicLibrary.cpp.o
> CMakeFiles/pdal_base.dir/pdal/EigenUtils.cpp.o
> CMakeFiles/pdal_base.dir/pdal/GDALUtils.cpp.o
> CMakeFiles/pdal_base.dir/pdal/GEOSUtils.cpp.o
> CMakeFiles/pdal_base.dir/pdal/Geometry.cpp.o
> CMakeFiles/pdal_base.dir/pdal/Kernel.cpp.o
> CMakeFiles/pdal_base.dir/pdal/Log.cpp.o
> CMakeFiles/pdal_base.dir/pdal/Metadata.cpp.o
> CMakeFiles/pdal_base.dir/pdal/Options.cpp.o
> CMakeFiles/pdal_base.dir/pdal/PDALUtils.cpp.o
> CMakeFiles/pdal_base.dir/pdal/PipelineExecutor.cpp.o
> CMakeFiles/pdal_base.dir/pdal/PipelineManager.cpp.o
> CMakeFiles/pdal_base.dir/pdal/PipelineReaderJSON.cpp.o
> CMakeFiles/pdal_base.dir/pdal/PipelineWriter.cpp.o
> CMakeFiles/pdal_base.dir/pdal/PluginDirectory.cpp.o
> CMakeFiles/pdal_base.dir/pdal/PluginManager.cpp.o
> CMakeFiles/pdal_base.dir/pdal/PointLayout.cpp.o
> CMakeFiles/pdal_base.dir/pdal/PointTable.cpp.o
> CMakeFiles/pdal_base.dir/pdal/PointView.cpp.o
> CMakeFiles/pdal_base.dir/pdal/Polygon.cpp.o
> CMakeFiles/pdal_base.dir/pdal/QuadIndex.cpp.o
> CMakeFiles/pdal_base.dir/pdal/Reader.cpp.o
> CMakeFiles/pdal_base.dir/pdal/Scaling.cpp.o
> CMakeFiles/pdal_base.dir/pdal/SpatialReference.cpp.o
> CMakeFiles/pdal_base.dir/pdal/Stage.cpp.o
> CMakeFiles/pdal_base.dir/pdal/StageExtensions.cpp.o
> CMakeFiles/pdal_base.dir/pdal/StageFactory.cpp.o
> CMakeFiles/pdal_base.dir/pdal/Streamable.cpp.o
> CMakeFiles/pdal_base.dir/pdal/Writer.cpp.o
> CMakeFiles/pdal_base.dir/pdal/XMLSchema.cpp.o
> CMakeFiles/pdal_base.dir/pdal/gitsha.cpp.o
> CMakeFiles/pdal_base.dir/pdal/pdal_config.cpp.o
> CMakeFiles/pdal_base.dir/pdal/compression/DeflateCompression.cpp.o
> CMakeFiles/pdal_base.dir/filters/private/DimRange.cpp.o
> CMakeFiles/pdal_base.dir/filters/private/Point.cpp.o
> CMakeFiles/pdal_base.dir/filters/private/Segmentation.cpp.o
> CMakeFiles/pdal_base.dir/filters/private/hexer/HexGrid.cpp.o
> CMakeFiles/pdal_base.dir/filters/private/hexer/Hexagon.cpp.o
> CMakeFiles/pdal_base.dir/filters/private/hexer/Path.cpp.o
> CMakeFiles/pdal_base.dir/filters/private/hexer/Processor.cpp.o
> CMakeFiles/pdal_base.dir/filters/private/hexer/Segment.cpp.o
> CMakeFiles/pdal_base.dir/kernels/private/density/OGR.cpp.o
> -Wl,-rpath,/var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/lib64:
> -lpthread -lgdal -lgeos_c -lgeotiff -lxml2 -lz -lcurl
> lib64/libpdal_util.so.8.0.0 lib64/libpdal_arbiter.a lib64/libpdal_kazhdan.a
> -ljsoncpp
>
> His suggestion to install curlpp hasn't resolved the issue for me, not
> even when I push the -lcurlpp into LDFLAGS:
>
> [ 53%] Linking CXX executable
> ../../bin/pdal_io_ilvis2_reader_metadata_test cd
> /var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/test/unit
> && /usr/bin/cmake -E cmake_link_script
> CMakeFiles/pdal_io_ilvis2_reader_metadata_test.dir/link.txt --verbose=1
> /usr/x86_64-pc-linux-gnu/gcc-bin/7.3.0/x86_64-pc-linux-gnu-g++
> -march=native -O2 -pipe -ggdb -Wl,-O1 -Wl,--as-needed -lcurlpp
> CMakeFiles/pdal_io_ilvis2_reader_metadata_test.dir/io/Ilvis2ReaderWithMDReaderTest.cpp.o
> CMakeFiles/pdal_test_support.dir/Support.cpp.o -o
> ../../bin/pdal_io_ilvis2_reader_metadata_test
> -Wl,-rpath,/var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/lib64
> ../../lib64/libpdal_base.so.8.0.0 ../../lib64/libpdal_util.so.8.0.0
> ../../lib64/libgtest.a -lpthread ../../lib64/libpdal_base.so.8.0.0:
> undefined reference to 'curl_easy_reset' ../../lib64/libpdal_base.so.8.0.0:
> undefined reference to 'curl_slist_append'
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to
> 'curl_easy_perform' ../../lib64/libpdal_base.so.8.0.0: undefined reference
> to 'curl_easy_init' ../../lib64/libpdal_base.so.8.0.0: undefined reference
> to 'curl_global_init' ../../lib64/libpdal_base.so.8.0.0: undefined
> reference to 'curl_easy_cleanup' ../../lib64/libpdal_base.so.8.0.0:
> undefined reference to 'curl_easy_setopt'
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to
> 'curl_easy_getinfo' ../../lib64/libpdal_base.so.8.0.0: undefined reference
> to 'curl_slist_free_all' collect2: error: ld returned 1 exit status
>
> I wonder if it can be a compiler issue.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PDAL/PDAL/issues/1822#issuecomment-443723644>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGLaSZWDD-MvD-iy2tCXFVNiKMIFhu08ks5u1TG1gaJpZM4SWcxg>
> .
>

I'm using Gentoo's build system. So I'm fairly certain it's looking into the right PATH.

[Complete build log](https://github.com/PDAL/PDAL/files/2639878/pdal-1.8.0-build.log.gz)

Now it gets interesting: Even when I inject -lcurl into LDFLAGS it's still not finding those symbols:
```
[ 53%] Linking CXX executable ../../bin/xml_schema_test
cd /var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/test/unit && /usr/bin/cmake -E cmake_link_script CMakeFiles/xml_schema_test.dir/link.txt --verbose=1
/usr/x86_64-pc-linux-gnu/gcc-bin/7.3.0/x86_64-pc-linux-gnu-g++  -march=native -O2 -pipe -ggdb  -Wl,-O1 -Wl,--as-needed -lcurl CMakeFiles/xml_schema_test.dir/XMLSchemaTest.cpp.o CMakeFiles/pdal_test_support.dir/Support.cpp.o  -o ../../bin/xml_schema_test -Wl,-rpath,/var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/lib64 ../../lib64/libpdal_base.so.8.0.0 ../../lib64/libpdal_util.so.8.0.0 ../../lib64/libgtest.a /usr/lib64/libxml2.so -lpthread 
../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_easy_reset'
../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_slist_append'
../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_easy_perform'
../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_easy_init'
../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_global_init'
../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_easy_cleanup'
../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_easy_setopt'
../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_easy_getinfo'
../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_slist_free_all'
collect2: error: ld returned 1 exit status
make[2]: *** [test/unit/CMakeFiles/xml_schema_test.dir/build.make:101: bin/xml_schema_test] Error 1
make[2]: Leaving directory '/var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build'
make[1]: *** [CMakeFiles/Makefile2:1389: test/unit/CMakeFiles/xml_schema_test.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Is the Curl Dev file installed ?

Le lun. 3 déc. 2018 à 15:49, Dirk <notifications@github.com> a écrit :

> I'm using Gentoo's build system. So I'm fairly certain it's looking into
> the right PATH.
>
> Complete build log
> <https://github.com/PDAL/PDAL/files/2639878/pdal-1.8.0-build.log.gz>
>
> Now it gets interesting: Even when I inject -lcurl into LDFLAGS it's still
> not finding those symbols:
>
> [ 53%] Linking CXX executable ../../bin/xml_schema_test
> cd /var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/test/unit && /usr/bin/cmake -E cmake_link_script CMakeFiles/xml_schema_test.dir/link.txt --verbose=1
> /usr/x86_64-pc-linux-gnu/gcc-bin/7.3.0/x86_64-pc-linux-gnu-g++  -march=native -O2 -pipe -ggdb  -Wl,-O1 -Wl,--as-needed -lcurl CMakeFiles/xml_schema_test.dir/XMLSchemaTest.cpp.o CMakeFiles/pdal_test_support.dir/Support.cpp.o  -o ../../bin/xml_schema_test -Wl,-rpath,/var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build/lib64 ../../lib64/libpdal_base.so.8.0.0 ../../lib64/libpdal_util.so.8.0.0 ../../lib64/libgtest.a /usr/lib64/libxml2.so -lpthread
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_easy_reset'
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_slist_append'
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_easy_perform'
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_easy_init'
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_global_init'
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_easy_cleanup'
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_easy_setopt'
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_easy_getinfo'
> ../../lib64/libpdal_base.so.8.0.0: undefined reference to `curl_slist_free_all'
> collect2: error: ld returned 1 exit status
> make[2]: *** [test/unit/CMakeFiles/xml_schema_test.dir/build.make:101: bin/xml_schema_test] Error 1
> make[2]: Leaving directory '/var/tmp/portage/sci-geosciences/pdal-1.8.0/work/pdal-1.8.0_build'
> make[1]: *** [CMakeFiles/Makefile2:1389: test/unit/CMakeFiles/xml_schema_test.dir/all] Error 2
> make[1]: *** Waiting for unfinished jobs....
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PDAL/PDAL/issues/1822#issuecomment-443735680>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGLaSYVq26zqrr2XiUPASGq_TioKvxKHks5u1Tn3gaJpZM4SWcxg>
> .
>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -274,7 +274,6 @@ target_link_libraries(${PDAL_BASE_LIB_NAME}
         ${ZLIB_LIBRARIES}
         ${LIBLZMA_LIBRARIES}
         ${ZSTD_LIBRARIES}
-        ${CURL_LIBRARIES}
         ${WINSOCK_LIBRARY}
         ${PDAL_REEXPORT}
         ${PDAL_UTIL_LIB_NAME}

